src/tablecloth/api/join_separate.clj

```diff
@@ -116,3 +118,68 @@
      (if (grouped? ds)       
        (process-group-data ds #(process-separate-columns % column target-columns replace-missing separator-fn drop-column?) parallel?)
        (process-separate-columns ds column target-columns replace-missing separator-fn drop-column?)))))
+
+
+(defn- prefix [keyword-or-string value]
+  (cond
+    (keyword? keyword-or-string) (keyword (str (name keyword-or-string) value))
+        (string? keyword-or-string) (str keyword-or-string value)))
```

Comment:
I think, the last case should be `:else`. There are many different column name types (like Symbol for example).
Also I think the hyphen should be added explicitely.

```clojure
(defn- prefix [prefix-name value]
  (let [with-prefix (str (->str prefix-name) "-" value)]
    (if (keyword? prefix-name)
      (keyword with-prefix)
      with-prefix)))
```

Reply:
what is function "->str" in your proposal ?

Reply:
I assume you mean "(name)".

In any case I took your proposal to add a dash by default, see last commit.